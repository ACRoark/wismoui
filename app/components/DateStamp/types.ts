export type DateStampFormat = 'auto' | 'long' | 'short';
