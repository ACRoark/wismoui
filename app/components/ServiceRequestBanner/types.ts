export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
