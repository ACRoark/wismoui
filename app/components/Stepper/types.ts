import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
