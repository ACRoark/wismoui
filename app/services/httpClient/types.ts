export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
