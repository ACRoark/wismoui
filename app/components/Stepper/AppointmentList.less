@import (reference) '../../styles/themes/index.less';

@appointment-list-cls-prefix: ~'@{dte-prefix}-appointment-list';

.@{appointment-list-cls-prefix} {
  margin-bottom: 10px;
}
