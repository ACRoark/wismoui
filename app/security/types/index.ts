import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
