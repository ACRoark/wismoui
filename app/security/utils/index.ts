import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
