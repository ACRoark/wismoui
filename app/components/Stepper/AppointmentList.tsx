import React, { FC, ReactElement } from 'react';

import useFlags from 'hooks/useFlags';
import { IAppointmentList, IServiceOrderEvent } from 'types';

import AppointmentDate1599 from './AppointmentDate1599';
import CompletedServiceOrderEvent from './CompletedServiceOrderEvent';
import mapAppointments from './mapAppointments';
import showAppointmentDate from './showAppointmentDate';

import './AppointmentList.less';

const AppointmentList: FC<IAppointmentList> = (props: IAppointmentList): ReactElement => {
  const { appointments, completed, date, showEditLink } = props;
  const { us3273 } = useFlags();
  const showDate = showAppointmentDate(appointments);

  if (completed) {
    return (
      <div className="dte-wismo-appointment-list">
        {appointments.map(
          (completedAppointment: IServiceOrderEvent): ReactElement => (
            <CompletedServiceOrderEvent
              date={completedAppointment.appointment.date}
              key={completedAppointment.product}
              product={completedAppointment.product}
            />
          ),
        )}
      </div>
    );
  }

  return (
    <div className="dte-wismo-appointment-list">
      {showDate && <AppointmentDate1599 date={date} showEditLink={showEditLink} />}
      {mapAppointments(appointments, us3273)}
    </div>
  );
};

export default AppointmentList;
