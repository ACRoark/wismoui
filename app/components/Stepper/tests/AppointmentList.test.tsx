import React from 'react';
import { Provider } from 'react-redux';
import { createMockStore, createSnapshotWithIntl } from 'testing/utils';

import { IServiceOrderEvent } from 'types';

import AppointmentList from '../AppointmentList';

const store = createMockStore();

describe('AppointmentList', (): void => {
  it('should return an appointment date and the appointment times when completed is false', (): void => {
    const appointments: IServiceOrderEvent[] = [
      {
        appointment: {
          date: '20200925',
          slotType: 'AM',
        },
        latestStatus: {
          createdAtDate: '20200922',
          createdAtTime: '140205',
          serviceOrderEventStatus: 'ON_SCHEDULE',
        },
        product: 'EFR_D1',
        serviceOrderCategory: 'SDI1',
        serviceOrderId: '800008161708',
      },
      {
        appointment: {
          date: '20200925',
          slotType: 'AM',
        },
        latestStatus: {
          createdAtDate: '20200922',
          createdAtTime: '140209',
          serviceOrderEventStatus: 'ON_SCHEDULE',
        },
        product: 'GFR_AS_H',
        serviceOrderCategory: 'SDI1',
        serviceOrderId: '800008161709',
      },
    ];

    const date = '20200925';

    const tree = createSnapshotWithIntl(
      <Provider store={store}>
        <AppointmentList
          appointments={appointments}
          completed={false}
          date={date}
          key={date}
          showEditLink={false}
        />
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });

  it('should return completed content when completed is true', (): void => {
    const appointments: IServiceOrderEvent[] = [
      {
        appointment: {
          date: '20200925',
          slotType: 'AM',
        },
        latestStatus: {
          createdAtDate: '20200922',
          createdAtTime: '140205',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
        serviceOrderCategory: 'SDI1',
        serviceOrderId: '800008161708',
      },
      {
        appointment: {
          date: '20200925',
          slotType: 'AM',
        },
        latestStatus: {
          createdAtDate: '20200922',
          createdAtTime: '140209',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'GFR_AS_H',
        serviceOrderCategory: 'SDI1',
        serviceOrderId: '800008161709',
      },
    ];

    const date = '20200925';

    const tree = createSnapshotWithIntl(
      <Provider store={store}>
        <AppointmentList
          appointments={appointments}
          completed
          date={date}
          key={date}
          showEditLink={false}
        />
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
