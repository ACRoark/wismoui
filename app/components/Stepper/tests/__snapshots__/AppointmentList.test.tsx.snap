// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentList should return an appointment date and the appointment times when completed is false 1`] = `
<div
  className="dte-wismo-appointment-list"
>
  <div
    className="dte-wismo-appointment-date"
  >
    09/25/20
  </div>
  <div>
    Electric appointment 
    8 a.m. - 12 p.m.
  </div>
  <div>
    Natural Gas appointment 
    8 a.m. - 12 p.m.
  </div>
</div>
`;

exports[`AppointmentList should return completed content when completed is true 1`] = `
<div
  className="dte-wismo-appointment-list"
>
  <span
    className="dte-wismo-completed-service-order-event"
  >
    Electric appointment completed on 
    09/25/20
  </span>
  <span
    className="dte-wismo-completed-service-order-event"
  >
    Natural Gas appointment completed on 
    09/25/20
  </span>
</div>
`;
